### PR TITLE
Use parallel hashmap in LS_CSR for edge data

### PR DESCRIPTION
**Pull request by meyerzinn**

### CODE REVIEW

**Review by patrickkenney9801 — APPROVED** [review]
Looks good to me, do things look like they work as expected?

File: libgalois/include/galois/graphs/LS_LC_CSR_Graph.h [hunk 1]
```diff
@@ -69,12 +69,13 @@ class LS_LC_CSR_Graph : private boost::noncopyable {
...
                          typename std::tuple<>>;
-  using EdgeDataStore = std::conditional_t<
-      HasEdgeData,
-      phmap::flat_hash_map<
-          std::pair<VertexTopologyID, VertexTopologyID>, EdgeData,
-          boost::hash<std::pair<VertexTopologyID, VertexTopologyID>>>,
-      std::tuple<>>;
+
+  using EdgeDataMap = phmap::parallel_flat_hash_map_m<
+      EdgeHandle, EdgeData, boost::hash<EdgeHandle>, std::equal_to<EdgeHandle>>;
```
Comment (patrickkenney9801) [inline libgalois/include/galois/graphs/LS_LC_CSR_Graph.h]: if you wanted you can probably drop the last 2 template args

File: libgalois/include/galois/graphs/LS_LC_CSR_Graph.h [hunk 2]
```diff
@@ -155,8 +155,10 @@ class LS_LC_CSR_Graph : private boost::noncopyable {
   }
 
   template <typename E = EdgeData, typename = std::enable_if<HasEdgeData>>
-  void setEdgeData(EdgeHandle handle, E data) {
-    m_edge_data[handle] = data;
+  inline void setEdgeData(EdgeHandle handle, E data) {
+    m_edge_data.lazy_emplace_l(
+        handle, [&](EdgeData& v) { v = data; },
```
Comment (patrickkenney9801) [inline libgalois/include/galois/graphs/LS_LC_CSR_Graph.h]: this should probably be `[&](std::pair<Key, Val>& v)` or explicitly use `map_type::value_type& v`

Comment (meyerzinn) [reply libgalois/include/galois/graphs/LS_LC_CSR_Graph.h]: `auto` to the rescue!

**Pull request merged**